desmod/simulation.py

```diff
@@ -220,13 +235,22 @@ def simulate_factors(base_config, factors, top_type,
     :param top_type: The model's top-level Component subclass.
     :param env_type: :class:`SimEnvironment` subclass.
     :param int jobs: User specified number of concurent processes.
+    :param int only_factor: User specified factor index to simulate.
     :returns: Sequence of result dictionaries for each simulation.
 
     """
-    configs = list(factorial_config(base_config, factors, 'meta.sim.special'))
+    configs = list(factorial_config(
+        base_config, factors, 'meta.sim.special', only_factor))
+    if only_factor is not None:
+        if len(configs) == 0:
+            raise ValueError('No factor at index {}'.format(only_factor))
+        assert len(configs) == 1, 'exactly one config expected when ' \
+            '--only_factor is used'
```

Comment:
The string associated with this assert presumes a `--only_factor` command line option, but that's not a valid assumption.